testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/ReadOnlyModeTestCase.java

```diff
@@ -0,0 +1,171 @@
+            Assert.assertTrue(Operations.readResult(masterClient.execute(Operations.createReadAttributeOperation(slaveAddress, "value"))).asBoolean());
+
+            // reload master HC
+            op = new ModelNode();
+            op.get(OP_ADDR).add(HOST, "master");
+            op.get(OP).set("reload");
+            domainMasterLifecycleUtil.executeAwaitConnectionClosed(op);
+            // Try to reconnect to the hc
+            domainMasterLifecycleUtil.connect();
+            domainMasterLifecycleUtil.awaitHostController(System.currentTimeMillis());
+
+            Assert.assertTrue(Operations.readResult(masterClient.execute(Operations.createReadAttributeOperation(domainAddress, "value"))).asBoolean());
+            Assert.assertTrue(Operations.readResult(masterClient.execute(Operations.createReadAttributeOperation(masterAddress, "value"))).asBoolean());
+            Assert.assertTrue(Operations.readResult(masterClient.execute(Operations.createReadAttributeOperation(slaveAddress, "value"))).asBoolean());
+
+            // reload slave HC
+            op = new ModelNode();
+            op.get(OP_ADDR).add(HOST, "slave");
+            op.get(OP).set("reload");
+            domainSlaveLifecycleUtil.executeAwaitConnectionClosed(op);
+            // Try to reconnect to the hc
+            domainSlaveLifecycleUtil.connect();
+            domainSlaveLifecycleUtil.awaitHostController(System.currentTimeMillis());
+
+            Assert.assertTrue(Operations.readResult(masterClient.execute(Operations.createReadAttributeOperation(domainAddress, "value"))).asBoolean());
+            Assert.assertTrue(Operations.readResult(masterClient.execute(Operations.createReadAttributeOperation(masterAddress, "value"))).asBoolean());
+            Assert.assertTrue(Operations.readResult(masterClient.execute(Operations.createReadAttributeOperation(slaveAddress, "value"))).asBoolean());
+        }
+        domainManager.stop();
+        domainConfig.getMasterConfiguration().setRewriteConfigFiles(false);
+        domainConfig.getSlaveConfiguration().setRewriteConfigFiles(false);
+        domainManager.getDomainMasterLifecycleUtil().startAsync();
+        domainManager.getDomainSlaveLifecycleUtil().startAsync();
+        try (final DomainClient clientMaster = getDomainClient(domainConfig.getMasterConfiguration())) {
+            waitForHostControllerBeingStarted(TIMEOUT_S, clientMaster);
+            Assert.assertTrue(Operations.getFailureDescription(clientMaster.execute(Operations.createReadAttributeOperation(domainAddress, "value"))).asString().contains("WFLYCTL0216"));
+            Assert.assertTrue(Operations.getFailureDescription(clientMaster.execute(Operations.createReadAttributeOperation(masterAddress, "value"))).asString().contains("WFLYCTL0216"));
+            Assert.assertTrue(Operations.readResult(clientMaster.execute(Operations.createReadAttributeOperation(slaveAddress, "value"))).asBoolean());
+        }
+    }
+
+
```

Comment:
Can you file a JIRA to make everything from here on down go away? This seems like something the DomainLifecycleUtil should handle. We've got a few bits and pieces like this scattered around the domain testsuite and it's the kind of thing that ends up causing trouble later.